File: app/src/main/java/vds/developer/aceworkout/Adapters/BottomNavPagerAdapter.java
package vds.developer.aceworkout.Adapters;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import vds.developer.aceworkout.fragments.ReportFragment;
import vds.developer.aceworkout.fragments.SettingFragment;
import vds.developer.aceworkout.fragments.SummaryFragment;
import vds.developer.aceworkout.fragments.TrainingFragment;

public class BottomNavPagerAdapter extends FragmentPagerAdapter {

    public BottomNavPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int fragmentIndex) {
        switch (fragmentIndex) {
            case 0: return new ReportFragment();
            case 1: return new SummaryFragment();
            case 2: return new TrainingFragment();
            case 3: return new SettingFragment();
            default: return new SummaryFragment();
        }
    }

    @Override
    public int getCount() {
        return 4;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        //to be configured in the activity
        return null;
    }


}

File: app/src/main/java/vds/developer/aceworkout/Adapters/BottomNavPagerAdapter.kt
package vds.developer.aceworkout.Adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import vds.developer.aceworkout.fragments.ReportFragment
import vds.developer.aceworkout.fragments.SettingFragment
import vds.developer.aceworkout.fragments.SummaryFragment
import vds.developer.aceworkout.trainingFragment.TrainingFragment

class BottomNavPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getItem(fragmentIndex: Int): Fragment {
        when (fragmentIndex) {
            0 -> return ReportFragment()
            1 -> return SummaryFragment()
            2 -> return TrainingFragment()
            3 -> return SettingFragment()
            else -> return SummaryFragment()
        }
    }

    override fun getCount(): Int {
        return 4
    }

    override fun getPageTitle(position: Int): CharSequence? {
        //to be configured in the activity
        return null
    }


}
